core/server/master/src/main/java/alluxio/master/file/meta/UfsBlockLocationCache.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.underfs.options.FileLocationOptions;
+
+import java.util.List;
+
+/**
+ * Cache for block locations in the UFS.
+ */
+public interface UfsBlockLocationCache {
+  /**
+   * Retrieves the block locations from UFS, and caches the result.
+   * If failed to get the locations from UFS, {@code null} is returned and nothing is cached.
+   * The result will overwrite the existing cached locations for the block.
+   *
+   * @param blockId the block ID
+   * @param fileUri the URI of the file which contains the block
+   * @param options the options for getting file locations from UFS
+   */
+  void process(long blockId, AlluxioURI fileUri, FileLocationOptions options);
```

Comment:
why is `process` separate from `get`? Wouldn't we just have a `get` method which will return the value in the cache if it exists, and otherwise, will do the process functionality transparently?

Reply:
originally, I did in this way, but afterwards, I think that complicates the signature of `get`.

Reply:
but I think it makes sense, I'll rename `process` to override `get`, which also returns the locations.